src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs

```diff
@@ -364,36 +630,81 @@ impl SharedBufferBatch {
     }
 }
 
+/// Iterate all the items in the shared buffer batch
+/// If there are multiple versions of a key, the iterator will return all versions
 pub struct SharedBufferBatchIterator<D: HummockIteratorDirection> {
     inner: Arc<SharedBufferBatchInner>,
-    current_idx: usize,
+    current_value_idx: i32,
```

Comment:
Any reason to adopt two separated idx for key and value? I think they are in the same progress even though we contains multiple epochs in the imm.

Reply:
After unifying the struct of merged_imm and imm, `SharedBufferBatch` becomes a two dimension array. The value index is for iterating different versions of a key on each `next()` call.

Reply:
The naming `value_idx` and `key_idx` is a little confused with the key and value for the iterator. Shall we have a less confusing naming?